src/connection/host_lookup.c

```diff
@@ -0,0 +1,444 @@
+//naming rule check, detailed rule, error set
+//if hostent_List[host_count].hostname is empty, set error
+      if (hostent_List[host_count].hostname != NULL)
+	{
+	  if ((user_host_Map.find (map_hostname) == user_host_Map.end ()))
+	    {
+	      user_host_Map[map_hostname] = map_ipaddr;
+	    }
+/*duplicated hostname*/
+	  else if (!strcmp (user_host_Map.find (map_hostname)->second.c_str (), hostent_List[host_count].ipaddr))
+	    {
+/*duplicated hostname and ip address both*/
+	    }
+	  else
+	    {
+/*duplicated hostname but different ip address*/
+//err_msg : duplicated 
+	      return NULL;
+	    }
+	  host_count++;
+	}
+    }
+  return host_count;
+}
+
+/*
+ * gethostbyname_uhost () - Do same job with gethostbyname (), using by the 'user' defined 'hosts.conf' file.
+ * 
+ * return   : the hostent pointer.
+ * hostname (in) : the hostname.
+ *
+ */
+struct hostent *
+gethostbyname_uhost (char *name)
+{
+  struct hostent *hp;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      return gethostbyname (name);
+    }
+  else
+    {
+      hp = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+    }
+
+  if (hp == NULL)
+    {
+      return NULL;
+    }
+
+  return hp;
+}
+
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+struct hostent *
+gethostbyname_r_uhost (const char *name, struct hostent *ret, char *buf, size_t buflen, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+int
+gethostbyname_r_uhost (const char *name, struct hostent *ret, struct hostent_data *ht_data)
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif /* HAVE_GETHOSTBYNAME_R_GLIBC */
+#else /* WINDOWS */
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#endif				/* HAVE_GETHOSTBYNAME_R */
+{
+#if defined (WINDOWS)
+  return 0;
+#endif
+
+  struct hostent *hp_buf = NULL;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+//err also modified by callee
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+      return gethostbyname_r (name, ret, buf, buflen, result, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+      return gethostbyname_r (name, ret, buf, buflen, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+      return gethostbyname_r (name, ret, &ht_data);
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif
+#endif
+    }
+  else
+    {
+      hp_buf = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+      memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+    }
+
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+  if (ret == NULL)
+    {
+      return EFAULT;
```

Comment:
```suggestion
      return ERANGE;
```

Reply:
ERANGE means below
"Math result not representable" or
"hostent buffer is small".

If you think EFAULT is not appropriate error, what about "EINVAL"?(Invalid argument)

Reply:
EINVAL is good.